test/e2e/setup.go

```diff
@@ -0,0 +1,411 @@
+	masterPrefix := randomAddressPrefix()
+	workerPrefix := randomAddressPrefix()
+	params := aznetwork.VirtualNetwork{
+		VirtualNetworkPropertiesFormat: &aznetwork.VirtualNetworkPropertiesFormat{
+			AddressSpace: &aznetwork.AddressSpace{
+				AddressPrefixes: &[]string{
+					"10.0.0.0/9",
+				},
+			},
+			Subnets: &[]aznetwork.Subnet{
+				{
+					SubnetPropertiesFormat: &aznetwork.SubnetPropertiesFormat{
+						AddressPrefix: to.StringPtr(masterPrefix),
+					},
+					Name: to.StringPtr(c.ClusterName + "-master"),
+				},
+				{
+					SubnetPropertiesFormat: &aznetwork.SubnetPropertiesFormat{
+						AddressPrefix: to.StringPtr(workerPrefix),
+					},
+					Name: to.StringPtr(c.ClusterName + "-worker"),
+				},
+			},
+		},
+		Name:     to.StringPtr(vnetName),
+		Type:     to.StringPtr("Microsoft.Network/virtualNetworks"),
+		Location: to.StringPtr(c.Location),
+	}
+
+	c.log.Infof("creating vnet")
+	err := c.virtualNetworks.CreateOrUpdateAndWait(ctx, c.VnetRG, vnetName, params)
+	if err != nil {
+		return err
+	}
+
+	rpClientID := "f1dd0a37-89c6-4e07-bcd1-ffd3d43d8875"
+	if os.Getenv("RP_MODE") == "development" {
+		rpClientID = os.Getenv("AZURE_FP_CLIENT_ID")
+	}
+	c.log.Debugf("assigning role to AZURE_FP_CLIENT_ID for vnet")
+	err = c.assignVnetRoleTo(ctx, rpClientID)
+	if err != nil {
+		return err
+	}
+	c.log.Debugf("assigning role to AZURE_CLUSTER_CLIENT_ID for vnet")
+	err = c.assignVnetRoleTo(ctx, c.ClusterClientID)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func parseTokenResponse(location string) (string, error) {
+	params := strings.Split(location, "#")[1]
+	for _, param := range strings.Split(params, "&") {
+		nameValue := strings.Split(param, "=")
+		if nameValue[0] == "access_token" {
+			return nameValue[1], nil
+		}
+	}
+	return "", fmt.Errorf("token not found in response")
+}
+
+func (c *ClientSet) getToken(apiServer, username, password string) (string, error) {
+	tokenURL, err := url.Parse(strings.Replace(apiServer, "api.", "oauth-openshift.apps.", 1))
+	if err != nil {
+		return "", err
+	}
+	tokenURL.Host = strings.Replace(tokenURL.Host, ":6443", "", 1)
+	tokenURL.Path = "/oauth/authorize"
+	q := tokenURL.Query()
+	q.Set("response_type", "token")
+	q.Set("client_id", "openshift-challenging-client")
+	tokenURL.RawQuery = q.Encode()
+	req, err := http.NewRequest("GET", tokenURL.String(), nil)
+	req.SetBasicAuth(username, password)
+	req.Header.Add("X-CSRF-Token", "1")
+
+	c.log.Debugf("url is %s", tokenURL.String())
+
+	resp, err := (&http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}).Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		c.log.Errorf(resp.Status)
+		return "", err
+	}
+
+	c.log.Debugf("resp %v", resp)
+	loc := resp.Header["Location"]
+	if loc == nil {
+		return "", fmt.Errorf("no Location header found") // this currently fails
```

Comment:
@jim-minter this Location is empty and is how we get the token. I am not sure how to craft the request to do the equivalent of "curl -L".
```
Test Panicked
  no Location header found
  /home/angus/go/src/github.com/Azure/ARO-RP/test/e2e/setup.go:389
```